crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -180,13 +189,14 @@ impl Mesh {
     /// Construct a new mesh. You need to provide a [`PrimitiveTopology`] so that the
     /// renderer knows how to treat the vertex data. Most of the time this will be
     /// [`PrimitiveTopology::TriangleList`].
-    pub fn new(primitive_topology: PrimitiveTopology) -> Self {
+    pub fn new(primitive_topology: PrimitiveTopology, cpu_persistent_access: bool) -> Self {
```

Comment:
What if `new` assumed `cpu_persistent_access = false`, and a second constructor, `new_with_persistence` (name chosen arbitrarily) was used for cases where you want to choose that value? It would reduce the amount of noise in this commit without hurting usability.

Reply:
The only thing I'm worried about with that is that it's non-obvious that the asset is going to disappear. I want users to be aware and consciously consider whether they need to keep the asset or not. 

I'm open to either option though.

Reply:
That is a good point. As long as it's for a good reason (which I think that concern is), the breaking change is worth having.